hazelcast/src/main/java/com/hazelcast/instance/impl/Node.java

```diff
@@ -331,15 +358,21 @@ public DiscoveryService createDiscoveryService(DiscoveryConfig discoveryConfig,
         }
         ILogger logger = getLogger(DiscoveryService.class);
 
+        final Map attributes = new HashMap<>(localMember.getAttributes());
+        if (properties.getBoolean(ClusterProperty.PERSISTENCE_AUTO_CLUSTER_STATE)
+            && config.getPersistenceConfig().isEnabled()) {
+            attributes.put(DISCOVERY_PROPERTY_THIS_NODE, this);
```

Comment:
good to have a test for these `DISCOVERY_PROPERTY_THIS_NODE` and `PERSISTENCE_AUTO_CLUSTER_STATE `?

Reply:
See `KubernetesTopologyIntentTrackerTest`.
I tried to test injection of the topology tracker in the discovery service via property but it seems to require running in kubernetes environment and I couldn't figure out a way to test it in a plain unit test.